src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -463,6 +465,55 @@ void program::set_options() {
     if (!options.get<build_option_type::force_implementations>()->forcing.empty()) {
         options.set_option(build_option::optimize_data(true));
     }
+
+    query_local_block_io_supported();
+}
+
+bool program::is_local_block_io_supported() const {
+    return is_subgroup_local_block_io_supported;
+}
+
+void program::query_local_block_io_supported() {
+    if (is_subgroup_local_block_io_supported == true)
```

Comment:
Maybe we can just check that device_info.supports_local_block_io == true to do actual compilation instead of adding state?

Reply:
It previously was. I will revert this to int.

Reply:
> Maybe we can just check that device_info.supports_local_block_io == true to do actual compilation instead of adding state?

Ok

Reply:
> Maybe we can just check that device_info.supports_local_block_io == true to do actual compilation instead of adding state?

It seems we need to state for 'not queried'. suppors_local_block_io is set to true if extension find "cl_intel_subgroup_local_block_io".